include/mpd/partition.h

```diff
@@ -0,0 +1,177 @@
+   PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+   PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+   LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+   NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+   SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*/
+
+/*! \file
+ * \brief MPD client library
+ *
+ * Do not include this header directly.  Use mpd/client.h instead.
+ */
+
+#ifndef MPD_PARTITION_H
+#define MPD_PARTITION_H
+
+#include "recv.h"
+#include "compiler.h"
+
+#include <stdbool.h>
+
+struct mpd_pair;
+/**
+ * \struct mpd_partition
+ */
+struct mpd_partition;
+struct mpd_connection;
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * Creates a new partition object from the pair received from MPD server.
+ *
+ * @param pair a #mpd_pair received from MPD (name must be "partition")
+ * @return the new #mpd_partition object, or NULL on error (out of
+ * memory, or pair name is not "partition")
+ *
+ * @since libmpdclient 2.17
+ */
+mpd_malloc
+struct mpd_partition *
+mpd_partition_new(const struct mpd_pair *pair);
+
+/**
+ * Frees a #mpd_partition object.
+ *
+ * @since libmpdclient 2.17
+ */
+void
+mpd_partition_free(struct mpd_partition *partition);
+
+/**
+ * Returns the channel name.
+ *
+ * @since libmpdclient 2.17
+ */
+mpd_pure
+const char *
+mpd_partition_get_channel(const struct mpd_partition *partition);
+
+/**
+ * Returns the partition name.
+ *
+ * @since libmpdclient 2.17
+ */
+mpd_pure
+const char *
+mpd_partition_get_partition(const struct mpd_partition *partition);
+
+/**
+ * Creates a new partition.
+ * A partition is one frontend of a multi-player MPD process: it has separate
+ * queue, player and outputs. A client is assigned to one partition at a time.
+ *
+ * @param connection the connection to MPD
+ * @param partition the partition name
+ * @return true on success
+ *
+ * @since libmpdclient 2.17
+ */
+bool
+mpd_send_newpartition(struct mpd_connection *connection, const char *partition);
+
+/**
+ * Shortcut for mpd_send_newpartition() and mpd_response_finish().
+ *
+ * @param connection the connection to MPD
+ * @param partition the partition name
+ * @return true on success
+ *
+ * @since libmpdclient 2.17
+ */
+bool
+mpd_run_newpartition(struct mpd_connection *connection, const char *partition);
+
+/**
+ * Switch the client to a different partition.
+ *
+ * @param connection the connection to MPD
+ * @param partition the partition name
+ * @return true on success
+ *
+ * @since libmpdclient 2.17
+ */
+bool
+mpd_send_partition(struct mpd_connection *connection, const char *partition);
+
+/**
+ * Shortcut for mpd_send_partition() and mpd_response_finish().
+ *
+ * @param connection the connection to MPD
+ * @param partition the partition name
+ * @return true on success
+ *
+ * @since libmpdclient 2.17
+ */
+bool
+mpd_run_partition(struct mpd_connection *connection, const char *partition);
```

Comment:
This command name may be confusion, unless you read the documentation.
Personally, i think mpd_switch_partition() or something would be better. However, in this case, it would not follow the MPD command name (which is just "partition"). I would also remove the mpd_send_partition() and keep just mpd_switch_partition()

Hence, i've left the same as the command name.

Reply:
There are a few examples of where libmpdclient's function names differ from the actual protocol names, because the MPD protocol is very terse, and libmpdclient shall provide a readable interface for programmers.
I'd agree with renaming the function to `mpd_run_switch_partition()`, but we also need `mpd_send_switch_partition()` (the variant of the function which just sends the command but doesn't wait/read for the response).

Reply:
Cool, ok. I've added a new commit for this.